src/topics.js

```diff
@@ -1,369 +1,373 @@
+				tags: async.apply(Topics.getTopicTagsObjects, tid),
+				isFollowing: async.apply(Topics.isFollowing, [tid], uid)
+			}, function(err, results) {
+				if (err) {
+					return callback(err);
+				}
+
+				topicData.posts = results.posts;
+				topicData.category = results.category;
+				topicData.thread_tools = results.threadTools.tools;
+				topicData.tags = results.tags;
+				topicData.isFollowing = results.isFollowing[0];
+
+				topicData.unreplied = parseInt(topicData.postcount, 10) === 1;
+				topicData.deleted = parseInt(topicData.deleted, 10) === 1;
+				topicData.locked = parseInt(topicData.locked, 10) === 1;
+				topicData.pinned = parseInt(topicData.pinned, 10) === 1;
+
+				plugins.fireHook('filter:topic.get', {topic: topicData, uid: uid}, function(err, data) {
+					callback(err, data ? data.topic : null);
+				});
+			});
+		});
+	};
+
+	function getMainPostAndReplies(topic, set, uid, start, stop, reverse, callback) {
+		async.waterfall([
+			function(next) {
+				posts.getPidsFromSet(set, start, stop, reverse, next);
+			},
+			function(pids, next) {
+				if ((!Array.isArray(pids) || !pids.length) && !topic.mainPid) {
+					return callback(null, []);
+				}
+
+				if (topic.mainPid) {
+					pids.unshift(topic.mainPid);
+				}
+				posts.getPostsByPids(pids, uid, next);
+			},
+			function(posts, next) {
+				if (!posts.length) {
+					return next(null, []);
+				}
+
+				if (topic.mainPid) {
+					posts[0].index = 0;
+				}
+
+				var indices = Topics.calculatePostIndices(start, stop, topic.postcount, reverse);
+				for (var i=1; i<posts.length; ++i) {
+					if (posts[i]) {
+						posts[i].index = indices[i - 1];
+					}
+				}
+
+				Topics.addPostData(posts, uid, callback);
+			}
+		]);
+	}
+
+	Topics.getMainPost = function(tid, uid, callback) {
+		Topics.getMainPosts([tid], uid, function(err, mainPosts) {
+			callback(err, Array.isArray(mainPosts) && mainPosts.length ? mainPosts[0] : null);
+		});
+	};
+
+	Topics.getMainPids = function(tids, callback) {
+		if (!Array.isArray(tids) || !tids.length) {
+			return callback(null, []);
+		}
+
+		Topics.getTopicsFields(tids, ['mainPid'], function(err, topicData) {
+			if (err) {
+				return callback(err);
+			}
+
+			var mainPids = topicData.map(function(topic) {
+				return topic && topic.mainPid;
+			});
+			callback(null, mainPids);
+		});
+	};
+
+	Topics.getMainPosts = function(tids, uid, callback) {
+		Topics.getMainPids(tids, function(err, mainPids) {
+			if (err) {
+				return callback(err);
+			}
+			getMainPosts(mainPids, uid, callback);
+		});
+	};
+
+	function getMainPosts(mainPids, uid, callback) {
+		posts.getPostsByPids(mainPids, uid, function(err, postData) {
+			if (err) {
+				return callback(err);
+			}
+			postData.forEach(function(post) {
+				if (post) {
+					post.index = 0;
+				}
+			});
+			Topics.addPostData(postData, uid, callback);
+		});
+	}
+
```

Comment:
I'd recommend you just `git reset HARD origin/master` file file, and re-add your function.